robotsim.py

```diff
@@ -48,118 +48,198 @@
+                start += 1
+            if distance == None:
+                distance = self.col
+
+        if self.dir == 2:
+            # row++ until max
+            for pos in range(self.row, map.height):
+                if map.tiles[pos][self.col].South.status in [1, 2]:
+                    distance = start
+                    break
+                start += 1
+            if distance == None:
+                distance = map.height - self.row - 1
+
+        if self.dir == 3:
+            # col++ until 0
+            for pos in range(self.col, map.width):
+                if map.tiles[self.row][pos].East.status in [1, 2]:
+                    distance = start
+                    break
+                start += 1
+            if distance == None:
+                distance = map.width - self.col - 1 
+        pygame.display.update()
+        clock.tick(120)
+        return distance
+
+    def detectSimbolLeft(self):
+        row = self.row
+        col = self.col
+        if self.dir == 0:
+            if map.tiles[row][col].West.status == 1:
+                return map.tiles[row][col].West.data
+        if self.dir == 1:
+            if map.tiles[row][col].South.status == 1:
+                return map.tiles[row][col].South.data
+        if self.dir == 2:
+            if map.tiles[row][col].East.status == 1:
+                return map.tiles[row][col].East.data
+        if self.dir == 3:
+            if map.tiles[row][col].North.status == 1:
+                return map.tiles[row][col].North.data
+        return None
+
+    def detectSimbolRight(self):
+        row = self.row
+        col = self.col
+        if self.dir == 0:
+            if map.tiles[row][col].East.status == 1:
+                return map.tiles[row][col].East.data
+        if self.dir == 1:
+            if map.tiles[row][col].North.status == 1:
+                return map.tiles[row][col].North.data
+        if self.dir == 2:
+            if map.tiles[row][col].West.status == 1:
+                return map.tiles[row][col].West.data
+        if self.dir == 3:
+            if map.tiles[row][col].South.status == 1:
+                return map.tiles[row][col].South.data
+        return None
+
+    def detectDoorFront(self):
+        row = self.row
+        col = self.col
+        if self.dir == 0 and map.tiles[row][col].North.status == 2:
+            return True
+        if self.dir == 1 and map.tiles[row][col].West.status == 2:
+            return True
+        if self.dir == 2 and map.tiles[row][col].South.status == 2:
+            return True
+        if self.dir == 3 and map.tiles[row][col].East.status == 2:
+            return True
+        return False
+
+    def insertCode(self, passw):
+        row = self.row
+        col = self.col
+        if self.dir == 0 and map.tiles[row][col].North.status == 2:
+            if map.tiles[row][col].North.data == passw:
+                map.tiles[row][col].North.status = 0
+                generate_map()
```

Comment:
In order to this to work, generate_map should be implemented using map.tiles, right now it is implemented with map.json and the wall remain.

Reply:
Add a TODO comment

Reply:
Done.